tests/system-tests/internal/ptp/ptp.go

```diff
@@ -0,0 +1,84 @@
+package ptp
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/pod"
+)
+
+func isPtPClockSync(apiClient *clients.Settings) (bool, error) {
```

Comment:
afaik timedatectl reports the status based on the NTP sync. For checking the PTP clock sync I think a better approach is to query the openshift_ptp_clock_state metric, e.g.

```
oc -n openshift-ptp exec linuxptp-daemon-h4kk4 -- curl -s http://localhost:9091/metrics | grep openshift_ptp_clock_state
Defaulted container "cloud-event-proxy" out of: cloud-event-proxy, kube-rbac-proxy, linuxptp-daemon-container
# HELP openshift_ptp_clock_state 0 = FREERUN, 1 = LOCKED, 2 = HOLDOVER
# TYPE openshift_ptp_clock_state gauge
openshift_ptp_clock_state{iface="CLOCK_REALTIME",node="sno.kni-qe-24.lab.eng.rdu2.redhat.com",process="phc2sys"} 1
openshift_ptp_clock_state{iface="ens1fx",node="sno.kni-qe-24.lab.eng.rdu2.redhat.com",process="ptp4l"} 1
```

Reply:
Great. I didn't know about it. 

I added it as an extra check. Thansk for the suggestion. Let me know if current approach is good enough.